core/src/apps/cardano/sign_tx.py

```diff
@@ -4,32 +4,42 @@
 from trezor.crypto import base58, hashlib
 from trezor.crypto.curve import ed25519
 from trezor.messages.CardanoSignedTx import CardanoSignedTx
-from trezor.messages.CardanoTxAck import CardanoTxAck
-from trezor.messages.CardanoTxRequest import CardanoTxRequest
 
-from apps.cardano import CURVE, seed
+from apps.cardano import CURVE, MAINNET_PROTOCOL_MAGIC, TESTNET_PROTOCOL_MAGIC, seed
 from apps.cardano.address import (
     derive_address_and_node,
     is_safe_output_address,
     validate_full_path,
 )
-from apps.cardano.layout import confirm_sending, confirm_transaction, progress
+from apps.cardano.layout import confirm_sending, confirm_transaction
 from apps.common import cbor
 from apps.common.paths import validate_path
 from apps.common.seed import remove_ed25519_prefix
 
+if False:
+    from typing import List, Tuple
+    from trezor.messages.CardanoSignTx import CardanoSignTx
+    from trezor.messages.CardanoTxInputType import CardanoTxInputType
+    from trezor.messages.CardanoTxOutputType import CardanoTxOutputType
+
 # the maximum allowed change address.  this should be large enough for normal
 # use and still allow to quickly brute-force the correct bip32 path
 MAX_CHANGE_ADDRESS_INDEX = const(1000000)
-ACCOUNT_PREFIX_DEPTH = const(2)
+ACCOUNT_PREFIX_DEPTH = const(3)
+
+LOVELACE_INVALID = 45_000_000_000 * 1_000_000
 
-KNOWN_PROTOCOL_MAGICS = {764824073: "Mainnet", 1097911063: "Testnet"}
+KNOWN_PROTOCOL_MAGICS = {
+    MAINNET_PROTOCOL_MAGIC: "Mainnet",
+    TESTNET_PROTOCOL_MAGIC: "Testnet",
+}
 
 
-# we consider addresses from the external chain as possible change addresses as well
-def is_change(output, inputs):
+# only addresses from the same account as the inputs are considered as change addresses
```

Comment:
I also don't like much that it is not consistent with what the HD derivation scheme says are change addresses (change addresses have specific value of the 4th item in the derivation path). So we may want to keep this behavior (not showing such outputs), but let's at least call the concept differently (e.g. this function could be `hide_output`) --- how are the Trezor people supposed to figure out why we are not following the details of HD specs?

Reply:
Yes, it would show all the outputs to the user. I think it's pretty harmless to show it to the user since he can simply check the address and this shouldn't even happen very often, right?

Reply:
I took another look at this and I don't really understand what this function is supposed to achieve. The previous comment:
`we consider addresses from the external chain as possible change addresses as well` kind of describes what it does. It's not about accounts, it's about external chains (a path different from `m/44'/1815'` e.g. `m/44'/1111'` will be treated as change address).

However when I try to use a change output with path `m/44'/1111'/0'/0/1` I get a "Forbidden key path" error. Which would mean that an address from the external chain can never occur in the change outputs.

Any ideas?